main.cpp

```diff
@@ -100,18 +106,45 @@ int main()
     /*
      construct 2 more Car instances via emplace_back.
      */
-    
+    cars.emplace_back("john");
+    cars.emplace_back("james");
+
     /*
      now reserve and emplace_back your Trucks and Motorcyles
      */
+    motorcycles.reserve(2);
+    trucks.reserve(2);
     
-    
-    
-    
-    assert(false);
+    motorcycles.emplace_back("paul");
+    motorcycles.emplace_back("matthew");    
+
+    trucks.emplace_back("luke");
+    trucks.emplace_back("thomas");  
+
+    //assert(false);
     //add the cars, motorcycles and trucks to the highway using range-based for() loops: for( element : vec ) { ... }
     //be careful to not accidentally make element copies when iterating.
-    
+
+    //  6) Add some Cars to the Highway
+
+    for( Car& car : cars ) 
+    {
+        highway.addVehicle(&car);
+    }
+
+    //  7) Add some Motorcycles to the Highway
+ 
+    for( Motorcycle& motorcycle : motorcycles ) 
+    {
+        highway.addVehicle(&motorcycle);
+    }
+
+    //  8) Add some SemiTrucks to the highway.
+    for( SemiTruck& semitruck : trucks ) 
+    {
+        highway.addVehicle(&semitruck);
+    }
```

Comment:
You can save some time by writing `for( auto& element : container )`.  If you change the type held by the container, you won't get error messages as you would currently with what is written.
- [ ] revise if you want.

Reply:
Cool ok done.